spring/boot-autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaAutoConfiguration.java

```diff
@@ -237,7 +238,12 @@ protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req)
 
         final Server s = server.build();
 
-        s.start().join();
+        s.start().handle((result, throwable) -> {
+            if (!Objects.isNull(throwable)) {
+                logger.error("Armeria server startup error: ", throwable);
+            }
+            return result;
```

Comment:
Doesn't this cause the code to continue, ignoring the error? How about instead of using `logger`, you just do something like `throw new IllegalStateException("Armeria server failed to start", t)` so the logging is added while still causing the exception to fail initialization?